tests/e2e/edge-middleware.test.ts

```diff
@@ -373,7 +373,11 @@ test.describe('RSC cache poisoning', () => {
   test('Middleware rewrite', async ({ page, middleware }) => {
     const prefetchResponsePromise = new Promise<Response>((resolve) => {
       page.on('response', (response) => {
-        if (response.url().includes('/test/rewrite-to-cached-page')) {
+        if (
+          (response.url().includes('/test/rewrite-to-cached-page') ||
+            response.url().includes('/caching-rewrite-target')) &&
```

Comment:
Note - the redirect that happens due to `validateRSCRequestHeaders` ( https://github.com/vercel/next.js/blob/21ff892d05e43644ebfd990d03d7f1742f9c46a7/packages/next/src/server/base-server.ts#L2086-L2098 ) result in seeing redirect with location being rewrite target